synapseclient/core/cache.py

```diff
@@ -281,22 +280,34 @@ def _cache_dirs(self):
                     if os.path.isdir(path2) and re.match('\\d+', item2):
                         yield path2
 
-    def purge(self, before_date, dry_run=False):
+    def purge(self, before_date=None, after_date=None, dry_run=False):
         """
         Purge the cache. Use with caution. Delete files whose cache maps were last updated prior to the given date.
 
         Deletes .cacheMap files and files stored in the cache.cache_root_dir, but does not delete files stored outside
         the cache.
         """
+        if before_date is None and after_date is None:
+            raise ValueError("Either before date or after date should be provided")
+
         if isinstance(before_date, datetime.datetime):
             before_date = utils.to_unix_epoch_time_secs(before_date)
+        if isinstance(after_date, datetime.datetime):
+            after_date = utils.to_unix_epoch_time_secs(after_date)
+
+        if before_date and after_date and before_date > after_date:
+            raise ValueError("Before date can not be larger than after date")
+
         count = 0
         for cache_dir in self._cache_dirs():
             # _get_modified_time returns None if the cache map file doesn't
             # exist and n > None evaluates to True in python 2.7(wtf?). I'm guessing it's
             # OK to purge directories in the cache that have no .cacheMap file
+
             last_modified_time = _get_modified_time(os.path.join(cache_dir, self.cache_map_file_name))
-            if last_modified_time is None or before_date > last_modified_time:
+            if last_modified_time is None or \
+                    (before_date and before_date > last_modified_time) or \
+                    (after_date and after_date < last_modified_time):
```

Comment:
I think maybe this should be:

```
if last_modified_time is None or (
                    (not before_date or before_date > last_modified_time) and 
                    (not after_date or after_date < last_modified_time)):
```

That way if both dates were supplied then anything between those dates would be deleted?